lib/google_sign_in_dartio.dart

```diff
@@ -29,37 +34,45 @@ typedef UrlPresenter = void Function(Uri uri);
 /// Implementation of the google_sign_in plugin in pure dart.
 class GoogleSignInDart extends platform.GoogleSignInPlatform {
   GoogleSignInDart._({
-    required DataStorage storage,
-    required String clientId,
-    required UrlPresenter presenter,
-    String? exchangeEndpoint,
-    int? port,
+    required final DataStorage storage,
+    required final String clientId,
+    required this.presenter,
+    final String? exchangeEndpoint,
+    final int? port,
+    final String? successUrl,
+    final String? failUrl,
   })  : _storage = storage,
         _clientId = clientId,
-        presenter = presenter,
         _port = port,
-        _exchangeEndpoint = exchangeEndpoint;
+        _exchangeEndpoint = exchangeEndpoint,
+        _successUrl = successUrl,
+        _failUrl = failUrl;
 
   /// Registers this implementation as default implementation for GoogleSignIn
   ///
   /// Your application should provide a [storage] implementation that can store
   /// the tokens is a secure, long-lived location that is accessible between
   /// different invocations of your application.
+  ///
+  /// Providing [successUrl] and [failUrl] will invoque such urls after
+  /// login succeeds or fails
   static Future<void> register({
-    required String clientId,
-    String? exchangeEndpoint,
+    required final String clientId,
+    final String? exchangeEndpoint,
     DataStorage? storage,
     UrlPresenter? presenter,
-    int? port,
+    final int? port,
```

Comment:
I think that we need to have an Options class that we should pass and not keep adding new values here.